applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c

```diff
@@ -257,43 +257,28 @@ static int modem_mode_request_cb(enum lte_lc_func_mode new_mode, void *user_data
 	return CONFIG_LWM2M_INTEGRATION_MODEM_MODE_REQUEST_RETRY_SECONDS;
 }
 
-static int firmware_update_state_cb(uint8_t update_state)
+static int lwm2m_firmware_event_cb(struct lwm2m_fota_event *event)
 {
-	int err;
-	uint8_t update_result;
 	struct cloud_wrap_event cloud_wrap_evt = { 0 };
 
-	/* Get the firmware object update result code */
-	err = lwm2m_get_u8(&LWM2M_OBJ(5, 0, 5), &update_result);
-	if (err) {
-		LOG_ERR("Failed getting firmware result resource value");
-		cloud_wrap_evt.type = CLOUD_WRAP_EVT_ERROR;
-		cloud_wrap_evt.err = err;
-		cloud_wrapper_notify_event(&cloud_wrap_evt);
-		return 0;
-	}
-
-	switch (update_state) {
-	case STATE_IDLE:
-		LOG_DBG("STATE_IDLE, result: %d", update_result);
-
-		/* If the FOTA state returns to its base state STATE_IDLE, the FOTA failed. */
-		cloud_wrap_evt.type = CLOUD_WRAP_EVT_FOTA_ERROR;
-		break;
-	case STATE_DOWNLOADING:
-		LOG_DBG("STATE_DOWNLOADING, result: %d", update_result);
+	switch (event->id) {
+	case LWM2M_FOTA_DOWNLOAD_START:
+		LOG_DBG("STATE_DOWNLOADING, instance: %d", event->download_start.obj_inst_id);
 		cloud_wrap_evt.type = CLOUD_WRAP_EVT_FOTA_START;
 		break;
-	case STATE_DOWNLOADED:
-		LOG_DBG("STATE_DOWNLOADED, result: %d", update_result);
+	/* FOTA download process finished */
```

Comment:
Can we have the comments inside the cases? That's typically how we have done it throughout the application.